experiments/long_runs/snowy_land.jl

```diff
@@ -0,0 +1,662 @@
+import ClimaParams as CP
+
+using ClimaLand
+using ClimaLand.Snow
+using ClimaLand.Soil
+using ClimaLand.Canopy
+import ClimaLand
+import ClimaLand.Parameters as LP
+
+using Statistics
+using CairoMakie
+import GeoMakie
+using Dates
+import NCDatasets
+
+const FT = Float64;
+time_interpolation_method = LinearInterpolation(PeriodicCalendar())
+regridder_type = :InterpolationsRegridder
+context = ClimaComms.context()
+device = ClimaComms.device()
+device_suffix = device isa ClimaComms.CPUSingleThreaded ? "cpu" : "gpu"
+root_path = "snowy_land_longrun_$(device_suffix)"
+diagnostics_outdir = joinpath(root_path, "global_diagnostics")
+outdir =
+    ClimaUtilities.OutputPathGenerator.generate_output_path(diagnostics_outdir)
+
+function setup_prob(t0, tf, Δt; outdir = outdir, nelements = (101, 15))
+
+    earth_param_set = LP.LandParameters(FT)
+    radius = FT(6378.1e3)
+    depth = FT(50)
+    domain = ClimaLand.Domains.SphericalShell(;
+        radius = radius,
+        depth = depth,
+        nelements = nelements,
+        npolynomial = 1,
+        dz_tuple = FT.((10.0, 0.05)),
+    )
+    surface_space = domain.space.surface
+    subsurface_space = domain.space.subsurface
+
+    start_date = DateTime(2021)
+    # Forcing data
+    era5_artifact_path =
+        ClimaLand.Artifacts.era5_land_forcing_data2021_folder_path(; context)
+    era5_ncdata_path = joinpath(era5_artifact_path, "era5_2021_0.9x1.25.nc")
+    atmos, radiation = ClimaLand.prescribed_forcing_era5(
+        era5_ncdata_path,
+        surface_space,
+        start_date,
+        earth_param_set,
+        FT;
+        time_interpolation_method = time_interpolation_method,
+        regridder_type = regridder_type,
+    )
+
+    soil_params_artifact_path =
+        ClimaLand.Artifacts.soil_params_artifact_folder_path(; context)
+    extrapolation_bc = (
+        Interpolations.Periodic(),
+        Interpolations.Flat(),
+        Interpolations.Flat(),
+    )
+    soil_params_mask = SpaceVaryingInput(
+        joinpath(
+            soil_params_artifact_path,
+            "vGalpha_map_gupta_etal2020_1.0x1.0x4.nc",
+        ),
+        "α",
+        subsurface_space;
+        regridder_type,
+        regridder_kwargs = (; extrapolation_bc,),
+        file_reader_kwargs = (; preprocess_func = (data) -> data > 0,),
+    )
+    oceans_to_value(field, mask, value) =
+        mask == 1.0 ? field : eltype(field)(value)
+
+    vg_α = SpaceVaryingInput(
+        joinpath(
+            soil_params_artifact_path,
+            "vGalpha_map_gupta_etal2020_1.0x1.0x4.nc",
+        ),
+        "α",
+        subsurface_space;
+        regridder_type,
+        regridder_kwargs = (; extrapolation_bc,),
+    )
+    vg_n = SpaceVaryingInput(
+        joinpath(
+            soil_params_artifact_path,
+            "vGn_map_gupta_etal2020_1.0x1.0x4.nc",
+        ),
+        "n",
+        subsurface_space;
+        regridder_type,
+        regridder_kwargs = (; extrapolation_bc,),
+    )
+    x = parent(vg_α)
```

Comment:
This might need to be `x = Array(parent(vg_α))` for GPU compatibility. On GPU, wrapping in `Array()` would force the `CuArray` returned by `parent` to be converted to an `Array`, so that operations on it happen on the CPU instead of GPU. Without this, I think we would get a scalar indexing error in line 133

Reply:
I dont think we actually need the parent...let me check

Reply:
not using parent would be ideal :)

Reply:
We do.... I think because if x is a Field, and we do x .> 0, it seems like it is trying to convert a field of booleans into a field of floats:

```
julia> typeof(x)
ClimaCore.Fields.Field{ClimaCore.DataLayouts.VIJFH{Float64, 15, 2, CUDA.CuArray{Float64, 5, CUDA.DeviceMemory}}, ClimaCore.Spaces.ExtrudedFiniteDifferenceSpace{ClimaCore.Grids.ExtrudedFiniteDifferenceGrid{ClimaCore.Grids.SpectralElementGrid2D{ClimaCore.Topologies.Topology2D{ClimaComms.SingletonCommsContext{ClimaComms.CUDADevice}, ClimaCore.Meshes.EquiangularCubedSphere{ClimaCore.Domains.SphereDomain{Float64}, ClimaCore.Meshes.NormalizedBilinearMap}, CartesianIndices{3, Tuple{Base.OneTo{Int64}, Base.OneTo{Int64}, Base.OneTo{Int64}}}, LinearIndices{3, Tuple{Base.OneTo{Int64}, Base.OneTo{Int64}, Base.OneTo{Int64}}}, CUDA.CuArray{Tuple{Int64, Int64, Int64, Int64, Bool}, 1, CUDA.DeviceMemory}, Vector{Tuple{Int64, Int64, Int64, Int64, Bool}}, CUDA.CuArray{Tuple{Int64, Int64}, 1, CUDA.DeviceMemory}, CUDA.CuArray{Int64, 1, CUDA.DeviceMemory}, CUDA.CuArray{Tuple{Bool, Int64, Int64}, 1, CUDA.DeviceMemory}, CUDA.CuArray{Int64, 1, CUDA.DeviceMemory}, CUDA.CuArray{Int64, 1, CUDA.DeviceMemory}, @NamedTuple{}, CUDA.CuArray{Tuple{Int64, Int64}, 1, CUDA.DeviceMemory}}, ClimaCore.Quadratures.GLL{2}, ClimaCore.Geometry.SphericalGlobalGeometry{Float64}, ClimaCore.DataLayouts.IJFH{ClimaCore.Geometry.LocalGeometry{(1, 2), ClimaCore.Geometry.LatLongPoint{Float64}, Float64, StaticArraysCore.SMatrix{2, 2, Float64, 4}}, 2, CUDA.CuArray{Float64, 4, CUDA.DeviceMemory}}, ClimaCore.DataLayouts.IJFH{Float64, 2, CUDA.CuArray{Float64, 4, CUDA.DeviceMemory}}, ClimaCore.DataLayouts.IFH{ClimaCore.Geometry.SurfaceGeometry{Float64, ClimaCore.Geometry.UVVector{Float64}}, 2, CUDA.CuArray{Float64, 3, CUDA.DeviceMemory}}, @NamedTuple{}}, FiniteDifferenceGrid{IntervalTopology{ClimaComms.SingletonCommsContext{ClimaComms.CUDADevice}, ClimaCore.Meshes.IntervalMesh{ClimaCore.Domains.IntervalDomain{ClimaCore.Geometry.ZPoint{Float64}, Tuple{Symbol, Symbol}}, Vector{ClimaCore.Geometry.ZPoint{Float64}}}, @NamedTuple{bottom::Int64, top::Int64}}, ClimaCore.Geometry.CartesianGlobalGeometry, ClimaCore.DataLayouts.VF{ClimaCore.Geometry.LocalGeometry{(3,), ClimaCore.Geometry.ZPoint{Float64}, Float64, StaticArraysCore.SMatrix{1, 1, Float64, 1}}, 15, CUDA.CuArray{Float64, 2, CUDA.DeviceMemory}}, ClimaCore.DataLayouts.VF{ClimaCore.Geometry.LocalGeometry{(3,), ClimaCore.Geometry.ZPoint{Float64}, Float64, StaticArraysCore.SMatrix{1, 1, Float64, 1}}, 16, CUDA.CuArray{Float64, 2, CUDA.DeviceMemory}}}, ClimaCore.Grids.Flat, ClimaCore.Geometry.ShallowSphericalGlobalGeometry{Float64}, ClimaCore.DataLayouts.VIJFH{ClimaCore.Geometry.LocalGeometry{(1, 2, 3), ClimaCore.Geometry.LatLongZPoint{Float64}, Float64, StaticArraysCore.SMatrix{3, 3, Float64, 9}}, 15, 2, CUDA.CuArray{Float64, 5, CUDA.DeviceMemory}}, ClimaCore.DataLayouts.VIJFH{ClimaCore.Geometry.LocalGeometry{(1, 2, 3), ClimaCore.Geometry.LatLongZPoint{Float64}, Float64, StaticArraysCore.SMatrix{3, 3, Float64, 9}}, 16, 2, CUDA.CuArray{Float64, 5, CUDA.DeviceMemory}}}, ClimaCore.Grids.CellCenter}}

julia> μ = mean(x[x .> 0]);
ERROR: Struct type Bool cannot be represented using base type Float64
Stacktrace:
  [1] error(s::String)
    @ Base ./error.jl:35
  [2] macro expansion
    @ ~/.julia/packages/ClimaCore/SvZXu/src/DataLayouts/struct.jl:34 [inlined]
  [3] check_basetype(::Type{Float64}, ::Type{Bool})
    @ ClimaCore.DataLayouts ~/.julia/packages/ClimaCore/SvZXu/src/DataLayouts/struct.jl:33
  [4] (ClimaCore.DataLayouts.VIJFH{Bool, 15, 2})(array::CUDA.CuArray{Float64, 5, CUDA.DeviceMemory})
    @ ClimaCore.DataLayouts ~/.julia/packages/ClimaCore/SvZXu/src/DataLayouts/DataLayouts.jl:918
  [5] similar(bc::Base.Broadcast.Broadcasted{ClimaCore.DataLayouts.VIJFHStyle{…}, Nothing, typeof(>), Tuple{…}}, ::Type{Bool}, ::Val{15})
    @ ClimaCore.DataLayouts ~/.julia/packages/ClimaCore/SvZXu/src/DataLayouts/broadcast.jl:377
  [6] similar(bc::Base.Broadcast.Broadcasted{ClimaCore.DataLayouts.VIJFHStyle{…}, Nothing, typeof(>), Tuple{…}}, ::Type{Bool})
    @ ClimaCore.DataLayouts ~/.julia/packages/ClimaCore/SvZXu/src/DataLayouts/broadcast.jl:364
  [7] similar(bc::Base.Broadcast.Broadcasted{…}, ::Type{…})
    @ ClimaCore.Fields ~/.julia/packages/ClimaCore/SvZXu/src/Fields/broadcast.jl:142
  [8] copy(bc::Base.Broadcast.Broadcasted{ClimaCore.Fields.FieldStyle{…}, ClimaCore.Spaces.ExtrudedFiniteDifferenceSpace{…}, typeof(>), Tuple{…}})
    @ ClimaCore.Fields ~/.julia/packages/ClimaCore/SvZXu/src/Fields/broadcast.jl:95
  [9] materialize(bc::Base.Broadcast.Broadcasted{ClimaCore.Fields.FieldStyle{…}, Nothing, typeof(>), Tuple{…}})
    @ Base.Broadcast ./broadcast.jl:867
 [10] top-level scope
    @ REPL[54]:1
Some type information was truncated. Use `show(err)` to see complete types.
```

but, that said, these scripts are running on GPU, so for some reason the parent call is OK:
```
julia> x = parent(vg_α);

julia> μ = mean(x[x .> 0]);

julia> typeof(x)
CUDA.CuArray{Float64, 5, CUDA.DeviceMemory}
```


so I will leave this as is